mrmustard/physics/representations.py

```diff
@@ -451,9 +451,17 @@ def __matmul__(self, other: Union[Bargmann, Fock]) -> Union[Bargmann, Fock]:
             )
 
         Abc = []
-        for A1, b1, c1 in zip(self.A, self.b, self.c):
-            for A2, b2, c2 in zip(other.A, other.b, other.c):
+        if settings.ELEMENT_WISE:
+            if self.ansatz.batch_size != other.ansatz.batch_size:
+                raise ValueError("Batch size of the two ansatze must match.")
```

Comment:
```suggestion
                raise ValueError(f"Batch size of the two ansatze must match since the settings.ELEMENT_WISE is  {settings.ELEMENT_WISE}.")
```